File: Backend/src/Fibertest30.Application/SystemEvent/EventData/OtdrTask/OtdrTaskProgressData.cs
using System.Text.Json;
namespace Fibertest30.Application;

public class OtdrTaskProgressData : ISystemEventData
{
    public string TaskId { get; }
    public string TaskType { get; }
    public int MonitoringPortId { get; }
    public string CreatedByUserId { get; }
    public int QueuePosition { get; }
    public string Status { get; }
    public double Progress { get; }
    public DateTime? CompletedAt { get; }
    public string StepName { get; }
    public string? FailReason { get; }

    public OtdrTaskProgressData(
        string taskId,
        OtdrTaskType taskType,
        int monitoringPortId, 
        string createdByUserId,
        int queuePosition, 
        string status, 
        double progress, 
        DateTime completedAt,
        string stepName, 
        string? failReason = null)
    {
        TaskId = taskId;
        TaskType = taskType.ToString();
        MonitoringPortId = monitoringPortId;
        CreatedByUserId = createdByUserId;
        QueuePosition = queuePosition;
        Status = status;
        Progress = progress;
        CompletedAt = completedAt == DateTime.MinValue ? null : completedAt;
        StepName = stepName;
        FailReason = failReason;
    }
    
    public string ToJsonData()
    {
        return JsonSerializer.Serialize(this);
    }
}

File: WebFt30/src/app/shared/system-events/system-event-data/otdr-task/otdr-task-progress-data.ts
export interface OtdrTaskProgressData {
  TaskId: string;
  TaskType: 'OnDemand' | 'Baseline' | 'Monitoring';
  MonitoringPortId: number;
  CreatedByUserId: string;
  QueuePosition: number;
  Status: 'Pending' | 'Cancelled' | 'Running' | 'Completed' | 'Failed';
  Progress: number;
  CompletedAt: string | null;
  StepName: string;
  FailReason: string;
}
